### PR TITLE
Add ASR task for SUPERB

### PR DESCRIPTION
**Pull request by lewtun**
This PR starts building up the SUPERB benchmark by including the ASR task as described in the [SUPERB paper](https://arxiv.org/abs/2105.01051) and `s3prl` [instructions](https://github.com/s3prl/s3prl/tree/master/downstream#asr-automatic-speech-recognition).

Usage:

```python
from datasets import load_dataset 

asr = load_dataset("superb", "asr")
# DatasetDict({
#     train: Dataset({
#         features: ['file', 'text', 'speaker_id', 'chapter_id', 'id'],
#         num_rows: 28539
#     })
#     validation: Dataset({
#         features: ['file', 'text', 'speaker_id', 'chapter_id', 'id'],
#         num_rows: 2703
#     })
#     test: Dataset({
#         features: ['file', 'text', 'speaker_id', 'chapter_id', 'id'],
#         num_rows: 2620
#     })
# })
```

I've used the GLUE benchmark as a guide for filling out the README.

To move fast during the evaluation PoC I propose to merge one task at a time, so we can continue building the training / evaluation framework in parallel.

Note: codewise this PR is ready for review - I'll add the missing YAML tags once #2620 is merged :)

### CODE REVIEW

**Comment by lewtun** [comment]
Wait until #2620 is merged before pushing the README tags in this PR

**Review by albertvillanova — COMMENTED** [review]
#2620 is already merged.

**Review by albertvillanova — COMMENTED** [review]
Thanks!

One question: aren't you adding `task_templates` to the `_info` method (and to the `dataset_infos.json`?

**Comment by lewtun** [comment]
> Thanks!
> 
> One question: aren't you adding `task_templates` to the `_info` method (and to the `dataset_infos.json`?

great catch! i've now added the asr task template (along with a mapping from superb task -> template) and updated the `dataset_infos.json` :)

File: datasets/superb/superb.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+def map_to_array(batch):
+    speech_array, _ = sf.read(batch["file"])
+    batch["speech"] = speech_array
+    return batch
+
+dataset = dataset.map(map_to_array, remove_columns=["file"])
+```
+"""
+
+TASK2TEMPLATE = {"asr": AutomaticSpeechRecognition(audio_file_path_column="file", transcription_column="text")}
```
Comment (albertvillanova) [inline datasets/superb/superb.py:100]: This dict might be removed.

Comment (albertvillanova) [inline datasets/superb/superb.py:100]:
```suggestion
```

File: datasets/superb/superb.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+
+class SuperbConfig(datasets.BuilderConfig):
+    """BuilderConfig for Superb."""
+
+    def __init__(
+        self,
+        data_url,
+        url,
```
Comment (albertvillanova) [inline datasets/superb/superb.py:109]:
```suggestion
        url,
        task_templates=None,
```

File: datasets/superb/superb.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        **kwargs,
+    ):
+        super(SuperbConfig, self).__init__(version=datasets.Version("1.9.0", ""), **kwargs)
+        self.data_url = data_url
+        self.url = url
```
Comment (albertvillanova) [inline datasets/superb/superb.py:114]:
```suggestion
        self.url = url
        self.task_templates = task_templates
```

File: datasets/superb/superb.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+                """\
+            ASR transcribes utterances into words. While PR analyzes the
+            improvement in modeling phonetics, ASR reflects the significance of
+            the improvement in a real-world scenario. LibriSpeech
+            train-clean-100/dev-clean/test-clean subsets are used for
+            training/validation/testing. The evaluation metric is word error
+            rate (WER)."""
+            ),
+            url="http://www.openslr.org/12",
+            data_url="http://www.openslr.org/resources/12/",
```
Comment (albertvillanova) [inline datasets/superb/superb.py:133]:
```suggestion
            data_url="http://www.openslr.org/resources/12/",
            task_templates=[AutomaticSpeechRecognition(audio_file_path_column="file", transcription_column="text")],
```

File: datasets/superb/superb.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+                    "text": datasets.Value("string"),
+                    "speaker_id": datasets.Value("int64"),
+                    "chapter_id": datasets.Value("int64"),
+                    "id": datasets.Value("string"),
+                }
+            ),
+            supervised_keys=("file", "text"),
+            homepage=self.config.url,
+            citation=_CITATION,
+            task_templates=[TASK2TEMPLATE[self.config.name]],
```
Comment (albertvillanova) [inline datasets/superb/superb.py:152]:
```suggestion
            task_templates=self.config.task_templates,
```

**Review by albertvillanova — COMMENTED** [review]
Good!

I have a suggested refactoring... Tell me what you think! :)

**Comment by lewtun** [comment]
> Good!
> 
> I have a suggested refactoring... Tell me what you think! :)

your approach is much more elegant - i've included your suggestions 🙏

**Review by albertvillanova — APPROVED** [review]
Thank you.

**Pull request merged**